### PR TITLE
added start_pipeline_execution operation

### PR DESCRIPTION
**Pull request by AdarShaked**
1.

### CODE REVIEW

File: cloudify_aws/codepipeline/resources/pipeline.py [hunk 1]
```diff
@@ -72,6 +74,20 @@ def delete(self, params=None):
                                             params=params))
         self.client.delete_pipeline(**params)
 
+    def execute(self, name=None, clientRequestToken=None):
+        """
+            start execution of an existing Pipeline.
+        """
+        params = {'name': name if name else self.resource_id}
+        if clientRequestToken:
+            params.update({"clientRequestToken": clientRequestToken})
```
Comment (EarthmanT) [inline cloudify_aws/codepipeline/resources/pipeline.py:83]: Is this a confidential asset?

File: cloudify_aws/codepipeline/resources/pipeline.py [hunk 2]
```diff
@@ -101,3 +117,11 @@ def create(ctx, iface, params, **_):
...
     iface.update_resource_id(resource_id)
     utils.update_resource_id(ctx.instance, resource_id)
+
+
+@operation
+@decorators.aws_resource(CodePipelinePipeline, RESOURCE_TYPE)
+def execute(ctx, iface, name=None, clientRequestToken=None, **_):
+    execute_response = iface.execute(name, clientRequestToken)
+    ctx.instance.runtime_properties[
+        'execute_pipeline_response'] = execute_response
```
Comment (EarthmanT) [inline cloudify_aws/codepipeline/resources/pipeline.py:127]: Any exceptions we need to handle?

Comment (AdarShaked) [reply cloudify_aws/codepipeline/resources/pipeline.py:127]: Well if it fails i got the client error with detailed message. I can add a catch that raises NonRecoverableError with traceback.

File: cloudify_aws/codepipeline/resources/pipeline.py (hunk 1 above)
Comment (AdarShaked) [reply cloudify_aws/codepipeline/resources/pipeline.py:83]: I didn't fully understood what this token used for and if it confidential. 
https://boto3.amazonaws.com/v1/documentation/api/1.12.13/reference/services/codepipeline.html#CodePipeline.Client.start_pipeline_execution
https://docs.aws.amazon.com/codepipeline/latest/APIReference/API_StartPipelineExecution.html

what do you think?

**Review by EarthmanT — APPROVED** [review]

**Review by EarthmanT — APPROVED** [review]

**Pull request merged**